From: Aristeu Rozanski <aris@redhat.com>
Date: Wed, 20 Aug 2008 13:16:50 -0400
Subject: [x86_64] nmi: setup apic to handle both IO APIC and LAPIC
Message-id: 20080820171647.487117000@redhat.com
O-Subject: [RHEL5.3 PATCH 04/25] nmi: modify setup_acpi_nmi_watchdog() to handle both IO APIC and LAPIC
Bugzilla: 447618

https://bugzilla.redhat.com/show_bug.cgi?id=447618

make setup_apic_nmi_watchdog() setup both IO APIC and LAPIC

upstream: f2802e7f571c05f9a901b1f5bd144aa730ccc88e (and others)

diff --git a/arch/x86_64/kernel/apic.c b/arch/x86_64/kernel/apic.c
index ca13cff..e2354f6 100644
--- a/arch/x86_64/kernel/apic.c
+++ b/arch/x86_64/kernel/apic.c
@@ -479,8 +479,7 @@ void __cpuinit setup_local_APIC (void)
 	}
 
 	nmi_watchdog_default();
-	if (nmi_watchdog == NMI_LOCAL_APIC)
-		setup_apic_nmi_watchdog();
+	setup_apic_nmi_watchdog();
 	apic_pm_activate();
 }
 
